vnext/Microsoft.ReactNative/Views/ScrollViewManager.cpp

```diff
@@ -121,6 +131,71 @@ void ScrollViewShadowNode::createView() {
       });
 }
 
+void ScrollViewShadowNode::OnPreviewKeyDown(
+    const winrt::IInspectable &sender,
+    const xaml::Input::KeyRoutedEventArgs &e) {
+  auto senderFrameworkElement = sender.try_as<winrt::FrameworkElement>();
+  auto xamlRoot = senderFrameworkElement.XamlRoot();
+  auto focusedElement =
+      xaml::Input::FocusManager::GetFocusedElement(xamlRoot).try_as<winrt::Windows::UI::Xaml::DependencyObject>();
+  auto findNextElementOptions = xaml::Input::FindNextElementOptions();
+  findNextElementOptions.SearchRoot(xamlRoot.Content());
+  auto previousKey = m_isHorizontal ? winrt::Windows::System::VirtualKey::Left : winrt::Windows::System::VirtualKey::Up;
+  auto nextKey = m_isHorizontal ? winrt::Windows::System::VirtualKey::Right : winrt::Windows::System::VirtualKey::Down;
+
+  if (e.Key() == previousKey) {
+    auto firstFocusableElement =
+        xaml::Input::FocusManager::FindFirstFocusableElement(senderFrameworkElement).try_as<xaml::DependencyObject>();
+    auto isFirstElementFocused = firstFocusableElement == focusedElement;
+    if (!isFirstElementFocused) {
+      xaml::Input::FocusManager::TryMoveFocus(xaml::Input::FocusNavigationDirection::Previous, findNextElementOptions);
+      e.Handled(true);
+    }
+  } else if (e.Key() == nextKey) {
+    auto lastFocusableElement =
+        xaml::Input::FocusManager::FindLastFocusableElement(senderFrameworkElement).try_as<winrt::DependencyObject>();
+    auto isLastElementFocused = lastFocusableElement == focusedElement;
+    if (!isLastElementFocused) {
+      xaml::Input::FocusManager::TryMoveFocus(xaml::Input::FocusNavigationDirection::Next, findNextElementOptions);
+      e.Handled(true);
+    }
+  } else if (e.Key() == winrt::Windows::System::VirtualKey::Tab) {
+    auto const &coreWindow = winrt::CoreWindow::GetForCurrentThread();
```

Comment:
needs an alternative for XAML Islands

Reply:
When will this break in Islands? We're using this code now, but only in 1 window.

Reply:
more than one [core]window per thread

Reply:
I see `winrt::CoreWindow::GetForCurrentThread()` being used in several places in this repo. Are those all potential bugs then?

Reply:
Yes, any getforcurrent {thread, view, window} are potential bugs. I had done a pass through to ensure we had meaningful fallbacks. At the same time, xaml islands isn't a scenario we _officially_ support for RNW (https://github.com/microsoft/react-native-windows-samples/issues/267), but if there is a scenario you are blocked on and there is a simple fix on the RNW side, we'd take it :)